File: Service/MatchService.cs
using System.Collections.Generic;
using Domain;
using projectC.Repository;

namespace projectC.Service
{
    public class MatchService
    {
        private readonly MatchRepository matchRepository;

        public MatchService(MatchRepository repository)
        {
            matchRepository = repository;
        }

        public List<Match> GetMatches() => matchRepository.GetAll();

        public void AddMatch(Match match) => matchRepository.Add(match);

        public void DeleteMatch(int id) => matchRepository.Remove(id);

        public void UpdateMatch(int id, Match match) => matchRepository.Update(id, match);

        public bool SellTickets(int matchId, int count) => matchRepository.SellTicketBySeats(matchId, count);
    }
}

File: project-java-BogdanBala112/HomeAssignment1/src/main/java/service/MatchService.java
package service;

import Entities.Match;
import Repository.MatchRepository;

import java.util.List;

public class MatchService {
    public MatchRepository matchRepository;

    public MatchService(MatchRepository matchRepository){
        this.matchRepository = matchRepository;
    }

    public List<Match> getMatches(){
        return matchRepository.getAll();
    }

    public void addMatch(Match match){
        matchRepository.add(match);
    }

    public void deleteMatch(int id){
        matchRepository.remove(id);
    }

    public void updateMatch(int id, Match match){
        matchRepository.update(id, match);
    }

    public boolean sellTickets(int matchId, int number){
        return matchRepository.sellTicketBySeats(matchId, number);
    }

}
